src/main/java/org/cryptomator/cryptofs/fh/OpenCryptoFile.java

```diff
@@ -70,15 +70,13 @@ public synchronized FileChannel newFileChannel(EffectiveOpenOptions options, Fil
 		try {
 			ciphertextFileChannel = path.getFileSystem().provider().newFileChannel(path, options.createOpenOptionsForEncryptedFile(), attrs);
 			initFileHeader(options, ciphertextFileChannel);
-			if (options.truncateExisting()) {
-				chunkCache.invalidateStale();
-				ciphertextFileChannel.truncate(cryptor.fileHeaderCryptor().headerSize());
-				fileSize.set(0);
-			}
 			initFileSize(ciphertextFileChannel);

```

Comment:
can be removed/replaced as well, can't it? The old ciphertextFileSize is most likely wrong at this point in time.

Reply:
It could be replaced, but it is not related to the issue.
The method initializes the file size variable on any _newly_ opened file. If the filesize is already initatilizes, the method directly returns.

So, if we opened the first time a file and truncate it, it does not matter because we are inside a synchronized block and no other thread can read wrong information.
If it is not the first time (aka fileSize is already initialized), the method does nothing, which is also okay.